torch_sparse/narrow.py

```diff
@@ -1,11 +1,69 @@
 from typing import Tuple
 
+import torch
 from torch_sparse.storage import SparseStorage
 from torch_sparse.tensor import SparseTensor
 
 
 def narrow(src: SparseTensor, dim: int, start: int,
-           length: int) -> SparseTensor:
+           length: int, bidim: False,
+           n_active_nodes: None) -> SparseTensor:
+    r"""It resizes :obj:`src` SparseTensor along one
+    specified dimension, or two dimensions at the same time
+    if the SparseTensor has 2 dimensions. In the
+    latter case, it returns a squared SparseTensor.
+    In the case of 1-dimensional resizing,
+    :obj:'start' and :obj:'lenght' are supported
+    In the case of bidimensional resizing, resizing
+    always starts from 0, and it returns a square matrix;
+    :obj:`src` must be a bidimensional SparseTensor
+    for bidim option to work.
+    Args:
+        src (SparseTensor): SparseTensor to be manipulated
+        dim (int): dimension along which the resizing is done.
+                    Not used in case bidim is True
+        start (int): first position to include in the ouput.
+                    Not used in case bidim is True (set to 0)
+        length (int): last position to include in the output.
+        bidim (bool):  it applies bidimensional and squared narrowing
+                    of the src, which must be a bidimensional SparseTensor.
+        n_active_nodes (int): Required only if bidim option is True;
+                    for efficient manipulation of the rowptr bidim case.
+    """
+    if bidim:
+        if len(src.storage._sparse_sizes) != 2:
+            raise NotImplementedError
+        
+        start = 0
+        if src.storage._rowptr is None:
+            rowptr, col, value = src.csr()
+
+        # rowptr
+        rowptr = torch.narrow(src.storage._rowptr,
+                              0, start, length + 1).clone()
+        rowptr[(n_active_nodes + 1):] = rowptr[n_active_nodes]
+
+        # col and value
+        col = torch.narrow(src.storage._col,
+                           0, start, rowptr[-1])
```

Comment:
Where do we filter `col` such that `col.max() < length`?

Reply:
This should an effect of the preliminary BFS ordering of the (batch) graph used to generate the input matrix `src`. 
BFS traversing of the batch graph generates an adj_transpose matrix which has, at least on the upper half, a sort of block diagonal structure while the last rows maybe rich of zeros, which should imply that the `col ` indices included in the row vector cut at `rowptr[-1]` are always less than `length`
This is specific of the trim_to_layer application. 
Please let me know if this does not make sense to you.

Reply:
OK! So I will continue the work on PyG only. Thanks for clarifying this.
I will then create a specific function for this resizing of the adj_transpose, with related test, and modify the `trim_to_layer ` function in PyG accordingly, in order to support `SparseTensor ` in `trim_to_layer`. 

If I get you right, I guess I can then close this PR.
I will open a draft PR on PyG soon.

Reply:
Sounds wonderful, thanks for your patience.